src/useLocalStore.ts

```diff
@@ -4,14 +4,19 @@ import React from "react"
 import { useAsObservableSourceInternal } from "./useAsObservableSource"
 import { isPlainObject } from "./utils"
 
+export function useLocalStore<TStore extends Record<string, any>>(initializer: () => TStore): TStore
 export function useLocalStore<TStore extends Record<string, any>, TSource extends object = any>(
```

Comment:
Wouldn't it be better to require `TSource` of object type here? I mean to remove default for `any` (which can be undefined).